master/buildbot/worker/base.py

```diff
@@ -77,6 +79,8 @@ def checkConfig(self, name, password, max_builds=None,
         @type locks: dictionary
         """
         self.name = name = ascii2unicode(name)
+        self.build_wait_timeout = build_wait_timeout
```

Comment:
you rather need to update this in reconfigureService(), or the reconfiguration will not work if build_wait_timeout is changed (better than nothing, though)